src/main/scala/tech/cryptonomic/conseil/tezos/PlatformDiscoveryOperations.scala

```diff
@@ -164,8 +125,85 @@ object PlatformDiscoveryOperations {
     distinctCount < maxCount
   }
 
-  /** Leaves only letters and digits in the SQL string */
+  /** Sanitizes string to be viable to paste into plain SQL */
   private def sanitizeForSql(str: String): String = {
-    str.filter(_.isLetterOrDigit)
+    str.filter(c => c.isLetterOrDigit || c == '_')
   }
+
+  /** Executes the query with given predicates
+    *
+    * @param  tableName name of the table which we query
+    * @param  query     query predicates and fields
+    * @return query result as a map
+    * */
+  override def queryWithPredicates(tableName: String, query: Query)(implicit ec: ExecutionContext): Future[List[Map[String, Any]]] = {
```

Comment:
Can we rewrite this function as a single line which is a composition of multiple smaller functions, all of which are descriptively named and individually unit tested?

Reply:
extracted one method, and added tests for both methods used here, `TezosDatabaseOperations.selectWithPredicates` already has tests